Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialComponent.cpp

```diff
@@ -59,7 +60,12 @@ namespace AZ
             BaseClass::Reflect(context);
             EditorMaterialComponentSlot::Reflect(context);
 
-            if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+            if (auto jsonContext = azrtti_cast<JsonRegistrationContext*>(context))
+            {
```

Comment:
So if I understand correctly - this is the branching point / key differentiation in serialization compared to before?

Reply:
Prefabs use the JSON serializer for everything that was previously written out with the object serializer in XML or binary. By default, there is a JSON serializer that will read and write everything in the serialize context. The system allows you to create custom serializers for specific types. In this case, I'm adding a custom serializer that will skip over the component data that's only needed for the property editor UI. Ideally that data would not need to be registered or persist at all but the edit context and RPE are built on top of the serialize context.